packages/webhid/src/methods.ts

```diff
@@ -69,13 +69,30 @@ export async function setupXkeysPanel(browserDevice: HIDDevice): Promise<XKeys>
 		})
 	})
 
-	// Wait for the device to initialize:
-	try {
-		await xkeys.init()
+	let alreadyRejected = false
+	return new Promise((resolve, reject) => {
+		const xkeysStopgapErrorHandler = (e: unknown) => {
+			if (alreadyRejected) {
+				console.error(`Xkeys: Error emitted after setup already rejected:`, e)
+				return
+			}
+			deviceWrap
+				.close()
+				.then(() => reject())
+				.catch(reject)
+				.finally(() => (alreadyRejected = true))
```

Comment:
doing this in the finally (old implementation) was not semantically coherent since the flag only makes sense when the promise rejects, not when it resolves (same applies below). So changed this.